include/simics/dmllib.h

```diff
@@ -2479,9 +2943,49 @@ _get_port_saved_variable(lang_void *saved_access, conf_object_t *_portobj,
                                   acc->dimension_sizes,
                                   acc->dimension_strides + portobj->ndims,
                                   acc->ndims,
+                                  _get_device_member_via_serializer,
                                   acc->getter);
 }
 
+UNUSED static uint32 *
+_DML_dimension_strides_from_dimsizes(const uint32 *dimsizes, uint32 dims,
```

Comment:
why not let the caller stack-allocate the stride array? It's small unless an adversary creates a thousand dimensions of size 1, in which case implicit index args are a bigger problem.

Reply:
The only reason is so the user wouldn't have to allocate the array. That's not much of a reason, admittedly, since this instead forces the user to do `MM_FREE`. I'll do stack-allocation.